CorsixTH/Src/th_map.h

```diff
@@ -182,6 +182,15 @@ enum class temperature_theme {
   yellow_red     //!< Gradients of yellow, orange, and red
 };
 
+enum tile_part : uint8_t {
+  ground = 0,
+  north_wall = 1,
+  west_wall = 2,
+  ui = 3,
+
+  _num_tile_parts = 4  //!< Number of parts in a tile.
```

Comment:
I'm not familiar with this leading underscore use in c++; did you source it from anywhere?

Reply:
I checked for a way to get the value from the enum declaration itself, but SO only listed highly convoluted solutions.

Settling for adding it as literal in the enumeration, I think it's not at equal footing with the other values, it's something more meta. So I wanted to make that clear in the name (which seems to be working :p ). As both Python and Lua have `_` prefixes for more private things, I thought it a good idea to do that here too.

Not that I care much about the precise name, dropped the leafing `_`.